### PR TITLE
Use new shared-ci script to build workers

### PR DESCRIPTION
**Pull request by paulbalaji**
#### Description
- replaced `build.sh` with `.shared-ci/scripts/build.sh`
- using `uploadAssemblies()` function in `.shared-ci/scripts/pinned-tools.sh` instead of manually setting `$ASSEMBLY_NAME` and calling toolbelt
- `pre-exit` --> `post-command` hook

#### Tests
ran through buildkite against `feature/unify-build-scripts` shared-ci branch

#### Documentation
n/a

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: ci/bootstrap.sh [hunk 1]
```diff
@@ -12,4 +12,4 @@ if [ -d "${SHARED_CI_DIR}" ]; then
     rm -rf "${SHARED_CI_DIR}"
 fi
 
-git clone "${CLONE_URL}" "${SHARED_CI_DIR}"
+git clone "${CLONE_URL}" "${SHARED_CI_DIR}" --branch feature/unify-build-scripts
```
Comment (paulbalaji) [inline ci/bootstrap.sh]: this change will be removed before the final merge

File: ci/launch.sh [hunk 2]
```diff
@@ -17,9 +15,7 @@ else
     ci/build-test.sh
 fi
 
-setAssemblyName "${PREFIX}"
-
-spatial cloud upload "${ASSEMBLY_NAME}" --log_level=debug --force --enable_pre_upload_check=false
+uploadAssembly "${ASSEMBLY_PREFIX}" "${PROJECT_NAME}"
```
Comment (jessicafalk) [inline ci/launch.sh]: will that work? I don't see you specifying either of the two variables anywhere

Comment (paulbalaji) [reply ci/launch.sh]: these are both defined in the buildkite step that calls this script

File: .buildkite/hooks/post-command [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+set -euo pipefail
```
Comment (samiwh) [inline .buildkite/hooks/post-command]: usual style is to separate these out.

Comment (samiwh) [reply .buildkite/hooks/post-command]: Do we not want -x

Comment (samiwh) [reply .buildkite/hooks/post-command]: ah just seen below. Do the two paths both get run?

Comment (paulbalaji) [reply .buildkite/hooks/post-command]: I kept the style from the pre-exit hook that we had before

**Pull request merged**